libbeat/reader/readfile/timeout.go

```diff
@@ -76,13 +76,21 @@ func (r *TimeoutReader) Next() (reader.Message, error) {
 		}()
 	}
 
+	timer := time.NewTimer(r.timeout)
+	defer func() {
+		timer.Stop()
+		select {
```

Comment:
This extra select looks redundant, is it still needed to free memory after `timer.Stop`? If so please explain in a comment

Reply:
You're right. Draining the channel is unnecessary unless the timer is going to be reused.

I took the opportunity to rewrite a little bit so that a single channel is used instead of allocating a new one every time.